### DISCUSSION TITLE
akregator: cannot open links in Firefox

### DISCUSSION THREAD

**Discussion by capmlbpkel** [netblue30/firejail#6519] [Q&A] [answered]
I would like to open links in Firefox when using Akregator, but this does not work.

Log:

```console
user@Z:~$ akregator
Reading profile /etc/firejail/akregator.profile
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-devel.inc
Reading profile /etc/firejail/disable-exec.inc
Reading profile /etc/firejail/disable-interpreters.inc
Reading profile /etc/firejail/disable-programs.inc
Reading profile /etc/firejail/disable-shell.inc
Reading profile /etc/firejail/whitelist-common.inc
Reading profile /etc/firejail/whitelist-run-common.inc
Reading profile /etc/firejail/whitelist-var-common.inc
Seccomp list in: !chroot, check list: @default-keep, prelist: unknown,
Parent pid 158911, child pid 158912
7 programs installed in 7.38 ms
Warning: not remounting /run/user/1000/doc
Seccomp list in: !chroot, check list: @default-keep, prelist: unknown,
Warning: cleaning all supplementary groups
Child process initialized in 140.53 ms
amdgpu: drmGetDevice2 failed.
libGL error: glx: failed to create dri3 screen
libGL error: failed to load driver: radeonsi
libGL error: failed to open /dev/dri/card0: No such file or directory
libGL error: failed to load driver: radeonsi
Connection to speech-dispatcher failed
Connection to speech-dispatcher failed
Connection to speech-dispatcher failed
Connection to speech-dispatcher failed
Connection to speech-dispatcher failed
Connection to speech-dispatcher failed
Connection to speech-dispatcher failed
Connection to speech-dispatcher failed
Connection to speech-dispatcher failed
$LIST_OF_FEEDS
kf.config.core: Couldn't write "/home/user/.config/akregatorrc" . Disk full?
"error occurred while parsing element" on line 1
QBuffer::seek: Invalid pos: 1441792
QBuffer::seek: Invalid pos: 1441792
kf.config.core: Couldn't write "/home/user/.config/akregatorrc" . Disk full?
```

I tried adding the following lines from `thunderbird.profile` to `$HOME/.config/firejail/akregator.local`:

```
noblacklist ${HOME}/.mozilla
whitelist ${HOME}/.mozilla/firefox/profiles.ini
read-only ${HOME}/.mozilla/firefox/profiles.ini
```

Unfortunately, that did not help. Then I added:

```
include firefox-common.profile
```

This made akregator start with a fresh profile without my feeds.

**Comment by kmk3** [comment, marked as answer]
There is no good way of doing this with just firejail at the moment.

Closing as a duplicate of:

* #6462

**Reply by capmlbpkel** [reply]
Thank you very much. If I understand the linked issue correctly, even if Firefox runs in a sandbox as well, links from akregator cannot be opened there since it is a different sandbox. Interestingly enough, opening links from Thunderbird in Firefox **does** work. Would it be feasible to transfer that approach to akregator?

**Reply by kmk3** [reply]
> Thank you very much.

No problem.

> If I understand the linked issue correctly, even if Firefox runs in a sandbox
> as well, links from akregator cannot be opened there since it is a different
> sandbox.

Yes, the main problem is the communication between sandboxes.

> Interestingly enough, opening links from Thunderbird in Firefox **does**
> work. Would it be feasible to transfer that approach to akregator?

From thunderbird.profile, it seems to use dbus to open links in Firefox:

```
dbus-user.talk org.mozilla.*
```

Which kind of "bypasses" the sandbox.

dbus does not appear to be blocked in akregator.profile, so if Akregator cannot
talk to Firefox, presumably it does not use dbus.

See also the workarounds in #6462.

**Reply by capmlbpkel** [reply]
Understood, thank you!

**Comment by kmk3** [comment]
(Offtopic)

Please see the following links for how to format code blocks in markdown:

* <https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks>
* <https://github.github.com/gfm/#fenced-code-blocks>
